common/ferrostar/src/navigation_controller/mod.rs

```diff
@@ -202,6 +214,21 @@ impl NavigationController {
                 let (current_step_geometry_index, snapped_user_location) =
                     self.snap_user_to_line(location, &current_step_linestring);
 
+                // Calculate distance increment between the user locations.
+                let distance_increment =
+                    distance_between_locations(&previous_user_location, &location) as u32;
+                let snapped_distance_increment = distance_between_locations(
+                    &previous_snapped_user_location,
+                    &snapped_user_location,
+                ) as u32;
+
+                // Update trip summary with accumulated distance
+                let updated_summary = self.update_trip_summary(
+                    summary,
+                    distance_increment,
+                    snapped_distance_increment,
+                );
+
```

Comment:
I *think* this would be a bit cleaner (esp as this function is already a long one) to move the distance increment calculation inside `update_trip_summary`.